src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs

```diff
@@ -449,9 +447,15 @@ private static async Task<RestoreSummary> PerformNuGetV2RestoreAsync(Common.ILog
 
                 string packagesConfigPath = GetPackagesConfigFilePath(pcRestoreMetadata.ProjectPath);
 
-                firstPackagesConfigPath = firstPackagesConfigPath ?? packagesConfigPath;
-
-                installedPackageReferences.AddRange(GetInstalledPackageReferences(packagesConfigPath, allowDuplicatePackageIds: true));
+                foreach (PackageReference packageReference in GetInstalledPackageReferences(packagesConfigPath))
+                {
+                    if (!packageReferenceToProjects.TryGetValue(packageReference, out List<string> value))
+                    {
+                        value ??= [];
```

Comment:
```suggestion
                        value ??= new();
```

Reply:
These are actually arrays/list, so I thought here it made sense.
Do you want these changed as well?

Reply:
I guess I don't understand this syntax then.  Does this say "set my list to an empty array"?  Or does this say "set my list to a new list"?  If it is the former, I would think that's bad since later on we call `.Add()` and the backing storage would need to be resized.  If it is the later, I prefer just `new()`.

Reply:
It's a new list/new array. 

I'll merge the suggestions.